set.mm

```diff
@@ -98434,6 +98434,345 @@ a Cantor normal form (and injectivity, together with coherence
       UM $.
   $}
 
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Transitive closure under a relationship
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c TrPred $.
+  $( Define the transitive predecessor class as a class. $)
+  ctrpred $a class TrPred ( R , A , X ) $.
+
+  ${
+    $d R a y $.  $d A a y $.  $d X a y $.
+    $( Define the transitive predecessors of a class ` X ` under a relationship
+       ` R ` and a class ` A ` .  This class can be thought of as the
+       "smallest" class containing all elements of ` A ` that are linked to
+       ` X ` by a chain of ` R ` relationships (see ~ trpredtr and
+       ~ trpredmintr ).  Definition based off of Lemma 4.2 of Don Monk's notes
```

Comment:
```suggestion
       ~ trpredmintr ).  Definition based on Lemma 4.2 of Don Monk's notes
```
("based off of" is too colloquial IMO and we should think of non-native speakers)